src/main/java/com/limechain/teavm/HttpRequest.java

```diff
@@ -1,8 +1,32 @@
 package com.limechain.teavm;
 
+import org.teavm.interop.Async;
+import org.teavm.interop.AsyncCallback;
 import org.teavm.jso.JSBody;
+import org.teavm.jso.JSFunctor;
+import org.teavm.jso.JSObject;
+import org.teavm.jso.core.JSError;
+
+import java.io.IOException;
 
 public class HttpRequest {
-    @JSBody(params = {"method", "url", "body"}, script = "return httpRequestSync(method, url, body);")
-    public static native String httpRequestSync(String method, String url, String body);
+    @JSFunctor
+    interface HttpRequestCallback extends JSObject {
+        void apply(JSError error, String response);
+    }
+
+    @Async
+    public static native String asyncHttpRequest(String method, String url, JSObject body);
+    private static void asyncHttpRequest(String method, String url, JSObject body, AsyncCallback<String> callback) {
+        createAsyncHttpRequest(method, url, body, (error, response) -> {
+            if (error != null) {
+                callback.error(new IOException(error.getMessage()));
```

Comment:
We have a bit of a circle jerk with the error here. The http.js class executes the following:

`callback(new Error('Request failed with status: ${response?.status}'), null);`

That ends up here. The `callback.error(exception)` which we call here returns the error back to the catch in the http.js, which on its own throws another error to the callback. Instead of doing `callback.error(exception)` here I suggest we use a logger to log the error in the console.

P.S. IOException isn't translated correctly to JS and we have a null for the message. If we ever have to throw an error back to the JS catch I suggest using Exception java error.